pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3759,3 +3305,112 @@ func getTaskRunStatus(t string, status corev1.ConditionStatus) *v1beta1.Pipeline
 		},
 	}
 }
+
+func buildTestData(
+	prs []*v1beta1.PipelineRun,
+	ps []*v1beta1.Pipeline,
+	ts []*v1beta1.Task,
+	clusterTasks []*v1beta1.ClusterTask,
+	trs []*v1beta1.TaskRun,
+	rs []*v1alpha1.PipelineResource,
+	conditions []*v1alpha1.Condition,
+) test.Data {
+	return test.Data{
+		PipelineRuns:      prs,
+		Pipelines:         ps,
+		TaskRuns:          trs,
+		Tasks:             ts,
+		ClusterTasks:      clusterTasks,
+		PipelineResources: rs,
+		Conditions:        conditions,
+	}
+}
+
+func getReconciledRun(
+	t *testing.T,
+	prs []*v1beta1.PipelineRun,
+	ps []*v1beta1.Pipeline,
+	ts []*v1beta1.Task,
+	clusterTasks []*v1beta1.ClusterTask,
+	trs []*v1beta1.TaskRun,
+	rs []*v1alpha1.PipelineResource,
+	conditions []*v1alpha1.Condition,
+	namespace, prName string,
+	wantEvents []string,
```

Comment:
minor: using newlines like this is not very standard in our Go codebase, we're much more likely to put these together on one line. ways to improve this:

* Docstring
* Long param lists are often identifed as a "code smell" (https://refactoring.guru/smells/long-parameter-list https://www.codingrevolution.com/long-parameter-list/#:~:text=Long%20parameter%20list%20in%20a,four%20is%20considered%20too%20many) the solution is usually to group these args together into an object you pass in - in this case I could see maybe instantiating a struct with these args and calling functions on it - that might help with some of the ambiguity around the many responsibilities of this function as well (if you wanted to break it up into multiple functions)